### PR TITLE
whatanime.ga command

### LINKED ISSUE #252: Integration with whatanime.ga | Anime detection
https://whatanime.ga/

It looks like it doesn't have an API but some kind of integration with the service would be a useful feature for the bot.

It doesn't need to necessarily be integrated with that site, but the idea still stands

### PR DESCRIPTION
**Pull request by pedrofracassi**
Closes #252 

Requires the `WHATANIME_API_KEY` environment variable.

### CODE REVIEW

**Comment by pedrofracassi** [comment]
image

File: src/commands/anime/whatanime.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  async run ({ t, author, channel, message }, urlFromParams) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    const imageUrl = urlFromParams || await lastImageFromChannel(channel)
+
+    if (imageUrl) {
+      if (new RegExp(/(http(s?):)([/|.|\w|\s|-])*\.(?:jpg|gif|png)/).exec(imageUrl)) {
+        const image = await snekfetch.get(imageUrl)
```
Comment (bolsomito) [inline src/commands/anime/whatanime.js]: `{ body }`

File: src/commands/anime/whatanime.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      }
+    } else {
+      embed.setTitle(t('commands:whatanime.noImage'))
+    }
+
+    channel.send(embed).then(() => channel.stopTyping())
+  }
+}
+
+async function lastImageFromChannel (channel) {
```
Comment (bolsomito) [inline src/commands/anime/whatanime.js]: you could make this a util

File: src/locales/en-US/commands.json [hunk 3]
```diff
@@ -351,5 +351,21 @@
   "hentai": {
     "commandDescription": "Sends a hentai in a NSFW channel",
     "hereIsYourHentai": "Here's your hentai!"
+  },
+  "whatanime": {
+    "commandDescription": "Tells you what anime an image came from. Gets the channel's last image if you don't give it any.",
+    "commandUsage": "[image url]",
```
Comment (bolsomito) [inline src/locales/en-US/commands.json]: <image | image url>

File: src/commands/anime/whatanime.js (hunk 2 above)
Comment (bolsomito) [reply src/commands/anime/whatanime.js]: (this = the function)

File: src/locales/en-US/commands.json (hunk 3 above)
Comment (pedrofracassi) [reply src/locales/en-US/commands.json]: but you don't type the image 🤔

**Comment by Doges** [comment]
you know

**Comment by Doges** [comment]
@pedrofracassi codacy

**Comment by pedrofracassi** [comment]
@Doges Codacy is glitchy. It passed travis, which uses the same package for testing.

**Pull request merged**